src/main/java/teammates/ui/controller/InstructorCourseStudentDetailsEditSaveAction.java

```diff
@@ -57,13 +57,21 @@ public ActionResult execute() throws EntityDoesNotExistException {
             
             boolean isSectionChanged = student.isSectionChanged(originalStudentAttribute);
             boolean isTeamChanged = student.isTeamChanged(originalStudentAttribute);
+            boolean isEmailChanged = student.isEmailChanged(originalStudentAttribute);
+            
             if (isSectionChanged) {
                 logic.validateSectionsAndTeams(Arrays.asList(student), courseId);
             } else if (isTeamChanged) {
                 logic.validateTeams(Arrays.asList(student), courseId);
             }
             
             logic.updateStudent(studentEmail, student);
+            
+            if (isEmailChanged) {
+                logic.resetStudentGoogleId(student.email, courseId);
+                taskQueuer.scheduleCourseRegistrationInviteToStudent(courseId, student.email, true);
```

Comment:
Wouldn't this send an email to student even before they are supposed to receive such emails i.e. even before any session is open